### DISCUSSION TITLE
Invalid traefik 2 websocket configuration?

### DISCUSSION THREAD

**Discussion by ProteanCode** [dani-garcia/vaultwarden#2391] [Q&A]
The proxy example at 
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

Shows a notation that looks like:
`- traefik.http.routers.bitwarden-websocket-http.rule=Host(`bitwarden.domain.tld`) && Path(`/notifications/hub`) && !Path(`/notifications/hub/negotiate`)`

But the traefik dashboard tells that this rule is an error 
`! is not supported`

**Comment by BlackDex** [comment]
Then it looks like you are using v1 maybe and not v2. Since the documentation of traefik clearly states it does. https://doc.traefik.io/traefik/routing/routers/#rule

**Reply by ProteanCode** [reply]
> Then it looks like you are using v1 maybe and not v2. Since the documentation of traefik clearly states it does. https://doc.traefik.io/traefik/routing/routers/#rule

https://github.com/traefik/traefik/issues/5228

Strange, I may have some old version of V2?

/ # traefik version
Version:      2.4.13
Codename:     livarot
Go version:   go1.16.6
Built:        2021-07-30T15:06:29Z
OS/Arch:      linux/amd64

**Comment by ProteanCode** [comment]
I have upgraded Traefik to v2.6
Now there is no webproxy error
I think its worth to note that v2.4 won't work with that rule

**Comment by BlackDex** [comment]
It's a wiki, so feel free to add it.
Looking at the changelogs that feature is there since v2.5.0.

**Comment by oreineke** [comment]
In the rule

 `Host(bitwarden.domain.tld) && Path(/notifications/hub) && !Path(/notifications/hub/negotiate)`,

I don't think it is necessary to include `!Path(/notifications/hub/negotiate)`, because `Path(/notifications/hub)` does not match the path `/notifications/hub/negotiate` anyway. 

Only `PathPrefix(/notifications/hub)` (note that is says "Patch*Prefix*" here) would match both `/notifications/hub` and `/notifications/hub/negotiate`.

See https://doc.traefik.io/traefik/routing/routers/#rule for details.
